### ISSUE TITLE
Precision issues in CPU Path, meshlights not working in OpenCL

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/LuxCore#290]
OS: Windows 10
LuxCore version: 32b117b1343285a3c64d3427a77865790421f946

This is a scene that shows some pretty strange problems: [weird_precision_bug.zip](https://github.com/LuxCoreRender/LuxCore/files/4085928/weird_precision_bug.zip)

It contains:
* two area lights
* a plane
* a cube and sphere that are instanced as a nested collection in Blender

A Path CPU render shows issues that look like floating point problems, however none of the objects are positioned at the world origin, and raising the min. epsilon does not help.

![cpu](https://user-images.githubusercontent.com/1584761/72723223-78fe3600-3b80-11ea-921b-3449bdee8435.png)

In the viewport with RT Path CPU, it looks like this:

![Capture](https://user-images.githubusercontent.com/1584761/72723488-0f325c00-3b81-11ea-9382-b9a7e628f67b.PNG)

Even weirder, on Path OpenCL, the image is black:

![ocl](https://user-images.githubusercontent.com/1584761/72723299-a8ad3e00-3b80-11ea-8557-33d5e1868685.png)

When a constantinfinite light is added, I get the following result, which seems to indicate to me that the area lights stop working when using Path OpenCL.

![ocl_with_constantinfinite](https://user-images.githubusercontent.com/1584761/72723423-eca04300-3b80-11ea-94a3-53adb74bdd45.png)

**Comment by Dade916** [comment]
Your test scene is working fine here:

![prec](https://user-images.githubusercontent.com/3247881/72727651-c3d17b00-3b8b-11ea-95e9-aa0248cd8f71.jpg)

?

**Comment by Theverat** [comment]
Interesting, when I run the exported scene in luxcoreui, it works for me as well.
Have you tried rendering in Blender? Maybe the problem is somehow circumvented when using FILESAVER engine?

**Comment by Dade916** [comment]
You are calling a Scene::DuplicateObject() on something that is already an instance. This will create a new instance with a different transformation while I have the feeling you are trying to create an instance of an instance, something not allowed in LuxCore.

The source of the problem is the definition of the nested object, if you delete the nested object in Blender (or comment all the code inside Scene::DuplicateObject()), it will work fine.

**Comment by Dade916** [comment]
I should have fixed the problem.

It was caused by a bug in the scene delete object procedure and triggered by the quite strange way the scene is built:

1) create an object (instanced);
2) duplicate the object (instanced in a different way);
3) erase the original object;

**Comment by Theverat** [comment]
Thanks. I guess I should still fix the issue in the addon, or is this no longer necessary?

**Comment by Theverat** [comment]
It's just how I implemented the instance export in the addon.
I have now changed it so the first exported instance is no longer part of the duplication list, and is no longer removed after the duplication. 

I'm not trying to create an instance of an instance, I only want to place one instance at some location in the world, and then place duplicates at other points in the world.

If it is OK to pass an instance to DuplicateObject, then everything's fine now.
